### PR TITLE
Remove feature flag for searchable snapshots

### LINKED ISSUE #7115: [Searchable Snapshots] Remove feature flag gating for GA release
**Is your feature request related to a problem? Please describe.**
Currently Searchable Snapshots feature is behind a feature flag. For Opensearch 2.7 release we are going GA for Searchable Snapshots . So, we have to remove Searchable Snapshots Feature flag before the 2.7 release.

**Describe the solution you'd like**
Remove the Searchable Snapshots feature flag created in https://github.com/opensearch-project/OpenSearch/pull/4680

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by kotwanikunal**
### Description
- Removes feature flag for searchable snapshots

### Issues Resolved
- Resolves #7115 

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff
- [x] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by andrross** [comment]
Looks like some real test failures here:


- [org.opensearch.backwards.MixedClusterClientYamlTestSuiteIT.test {p0=snapshot.restore/10_basic/Create a snapshot and then restore it}](https://build.ci.opensearch.org/job/gradle-check/13914/testReport/junit/org.opensearch.backwards/MixedClusterClientYamlTestSuiteIT/test__p0_snapshot_restore_10_basic_Create_a_snapshot_and_then_restore_it_/)
- [org.opensearch.cluster.ClusterModuleTests.testAllocationDeciderOrder](https://build.ci.opensearch.org/job/gradle-check/13914/testReport/junit/org.opensearch.cluster/ClusterModuleTests/testAllocationDeciderOrder/)
- [org.opensearch.node.NodeTests.testCreateWithFileCache](https://build.ci.opensearch.org/job/gradle-check/13914/testReport/junit/org.opensearch.node/NodeTests/testCreateWithFileCache/)

File: server/src/main/java/org/opensearch/action/admin/cluster/snapshots/restore/RestoreSnapshotRequest.java [hunk 1]
```diff
@@ -146,7 +145,7 @@ public RestoreSnapshotRequest(StreamInput in) throws IOException {
         indexSettings = readSettingsFromStream(in);
         ignoreIndexSettings = in.readStringArray();
         snapshotUuid = in.readOptionalString();
-        if (FeatureFlags.isEnabled(FeatureFlags.SEARCHABLE_SNAPSHOT) && in.getVersion().onOrAfter(Version.V_2_4_0)) {
+        if (in.getVersion().onOrAfter(Version.V_2_4_0)) {
```
Comment (andrross) [inline server/src/main/java/org/opensearch/action/admin/cluster/snapshots/restore/RestoreSnapshotRequest.java:148]: Due to the feature flag, 2.4 doesn't actually understand this field (by default). I think we'll have to change this to 3.0 in this PR, then change it to 2.7 after the backport.

Comment (andrross) [reply server/src/main/java/org/opensearch/action/admin/cluster/snapshots/restore/RestoreSnapshotRequest.java:148]: It does mean the released version will be incompatible with the pre-release feature-flagged versions, which I think is probably fine.

Comment (kotwanikunal) [reply server/src/main/java/org/opensearch/action/admin/cluster/snapshots/restore/RestoreSnapshotRequest.java]: Yep. I think we also need to update the `V_3_0_0` references with `V_2_7_0` later to have the correct BWC.

**Comment by andrross** [comment]
Created issue #7128 for flaky failures in https://build.ci.opensearch.org/job/gradle-check/13937/

**Comment by andrross** [comment]
We should include a changelog entry announcing the feature as a part of this PR. Probably link to [the meta issue](#5087).

**Review by andrross — APPROVED** [review]

**Comment by kotwanikunal** [comment]
I'll take care of the backport.

**Pull request merged**